File: solutions/exercise-06/src/main/java/io/miragon/training/application/service/RejectMembershipService.java
package io.miragon.training.application.service;

import io.miragon.training.application.port.inbound.RejectMembershipUseCase;
import io.miragon.training.application.port.outbound.MembershipProcess;
import io.miragon.training.domain.MembershipId;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class RejectMembershipService implements RejectMembershipUseCase {

    private static final Logger log = LoggerFactory.getLogger(RejectMembershipService.class);

    private final MembershipProcess process;

    public RejectMembershipService(MembershipProcess process) {
        this.process = process;
    }

    @Override
    public void rejectMembership(MembershipId membershipId) {
        log.info("Rejecting membership confirmation for membershipId={}", membershipId.value());
        process.rejectMembership(membershipId);
    }
}

File: solutions/exercise-5/src/main/kotlin/io/miragon/training/application/service/RejectMembershipService.kt
package io.miragon.training.application.service

import io.github.oshai.kotlinlogging.KotlinLogging
import io.miragon.training.application.port.inbound.RejectMembershipUseCase
import io.miragon.training.application.port.outbound.MembershipProcess
import io.miragon.training.domain.MembershipId
import org.springframework.stereotype.Service

@Service
class RejectMembershipService(
    private val process: MembershipProcess,
) : RejectMembershipUseCase {

    private val log = KotlinLogging.logger {}

    override fun rejectMembership(membershipId: MembershipId) {
        log.info { "Rejecting membership confirmation for membershipId=${membershipId.value}" }
        process.rejectMembership(membershipId)
    }
}
